### PR TITLE
Minor documentation fixes for 0.32 release

### PR DESCRIPTION
**Pull request by mudit2812**
**Description of the Change:**
* Added example for qubit reset/reuse to `measurements.rst`
* Fixed docstring for `QuantumScript.bind_new_parameters`
* Fixed docstring for `spsa_grad`
* Added not about using state prep operator mid-circuit to `StatePrep`, `BasisState`, and `QutritBasisState`
* Fixed docstrings of functions in `qchem/convert.py`

### CODE REVIEW

**Review by timmysilv — APPROVED** [review]

File: pennylane/qchem/convert.py [hunk 1]
```diff
@@ -414,7 +414,7 @@ def _excited_configurations(electrons, orbitals, excitation):
     r"""Generate excited configurations from a Hartree-Fock reference state.
 
     This function generates excited configurations in the form of integers representing a binary
-    string, e.g., :math:`|1 1 0 1 0 0 \rangle is represented by :math:`int('110100', 2) = 52`.
+    string, e.g., :math:`|1 1 0 1 0 0 \rangle is represented by ``int('110100', 2) = 52``.
```
Comment (Jaybsoni) [inline pennylane/qchem/convert.py:417]: This looks weird ... Are we sure that we want "is represented" to render as math text as well? I would intuitively expect the fix to be to add a quote after \rangle. Feel free to resolve this if that still doesn't fix the issue.

```suggestion
    string, e.g., :math:`|1 1 0 1 0 0 \rangle` is represented by :math:`int('110100', 2) = 52`.
```

**Review by Jaybsoni — APPROVED** [review]
LGTM 💯

File: doc/introduction/measurements.rst [hunk 2]
```diff
@@ -309,6 +309,26 @@ Note that we can also specify an outcome when defining a conditional operation:
...
+        qml.PauliX(1)
+        m_0 = qml.measure(1, reset=True)
+        qml.PauliX(1)
+        return qml.probs(wires=[1])
+
+Executing this QNode:
+
+>>> func()
+tensor([0., 1.], requires_grad=True)
+
```
Comment (trbromley) [inline doc/introduction/measurements.rst:331]: Thanks!

File: pennylane/ops/qubit/state_preparation.py [hunk 3]
```diff
@@ -132,6 +137,11 @@ class StatePrep(StatePrepBase):
         using the method developed by Möttönen et al. (Quantum Info. Comput.,
         2005).
 
+    .. note::
+
+        When called in the middle of a circuit, the action of the operation is defined
+        as :math:`U|0\rangle = |\psi\rangle`
+
```
Comment (trbromley) [inline pennylane/ops/qubit/state_preparation.py:144]: Awesome!

**Pull request merged**